content/articles/storing-data-in-python-using-json-module/index.md

```diff
@@ -0,0 +1,166 @@
+
+We then access the file in `'w'`mode(*write mode*), to make JSON capable of writing data to a file. Finally,the `json.dump()` function keeps the set of numbers in the file *numbers.json*
+
+This program has no terminal output, but when we open the file *numbers.json* we see the following data:
+
+```
+[10, 20, 30, 70, 191, 23]
+```
+
+### Using json.load()
+Let's construct a program that reads the set of numbers back into memory using `json.load()`, considering the previous example:
+
+```python
+import json
+
+filename = 'numbers.json'  #make sure is the same file you wrote to  
+with open(filename,'r') as file_object:  #open in read mode
+ numbers = json.load(file_object)#use json.load to load data and store it in the variable numbers
+
+print(numbers)     #print the set of numbers
+```
+
+The `json.load(`) function takes one argument which is the file object. 
+
+Ensure that you read from the same file that you wrote to. Python only requires to read out of that file hence open it in *read mode*.
+
+In the above program, the `json.load()` function to loads the
+information stored in *numbers.json*, and then store it in the variable *numbers*. Then we print the set of numbers:
+
+```
+[10, 20, 30, 70, 191, 23]
+```
+
+This is a simple technique for two programs to share data.
+
+
+### Saving and reading user-generated data
+
+When dealing with data provided by users, storing data with JSON is helpful since if you don't save your user's data in some way, you'll lose everything when your program terminates.
+
+Consider the following case: we ask the user for their first name when they execute our program for the first time, and then
+remember their first name when they re-run our program.
+
+First, let's save their first name:
+
+```python
+import json
+
+firstname = input("Enter your first name to register? ")
+filename = 'firstname.json'
+with open(filename, 'w') as file_obj:
+     json.dump(firstname, file_obj)
+     print("Thank you for registering in our community " + firstname + "!")
+```
+
+In this program, we ask for the first name to register the user.
+
+We use `json.dump()` to store the first name in the file *firstname.json*. 
+
+The user will then see a notification stating that he or she has been registered.
+
+```
+Enter your first name to register? Felix
+Thank you for registering in our community Felix!
+```
+
+Let's now create a new program that welcomes a user whose first name has been registered.
+
+```python
+import json
+
+filename = 'firstname.json'
+with open(filename,'r') as file_obj:
+ firstname = json.load(file_obj)
+ print("Welcome back, " + firstname + " Thank you for registering!")
+ ```
+
+Here we use the `json.load()` to read the data saved in *firstname.json* into the variable *firstname*. We can finally welcome them back now that we've recovered their first name:
+
+```
+Welcome , Felix  Thank you for registering!
+```
```

Comment:
You are doing the same thing here.

Though, you are accepting user info from the user doesn't make this a new program. You can even do the same in your very first snippet too.